wandb/sdk/wandb_login.py

```
@@ -185,9 +180,8 @@ def configure_api_key(self, key: str) -> None:
                 "WANDB_API_KEY environment variable, or running "
                 "`wandb login` from the command line."
             )
-        apikey.write_key(self._settings, key)
-        self.update_session(key)
-        self._key = key
+        if key:
```

what happens if we call configure_api_key without a key is it just silently noop?

---

Yeah this would be a noop, which is a good point we can let the key pass to `write_key` and if the key is false-y then an error is thrown.

In actuality, this should never happen. Consider 3 cases
1. the user already has a key configured (without `--relogin`) then `if key -> True`
2. The user passes the key as an arg `wandb login {key}` -> then `if key -> True`
3. The users key is not configured (or `--relogin` is provided) and the key is not an arg.
  In this case we will continually re-prompt for the key until it is not false-y

---

so should we just raise an exception since we expect the api respect providing a string apikey?

---

Yes we should raise an exception

---

Actually I remember why this case exists. We can get an empty key during the login prompt timeout. 
The current experience is that the setting for `mode = disabled`, the check for key is to keep this experience.